examples/generative/Density estimation using RealNVP.py

```diff
@@ -0,0 +1,217 @@
+
+requirements:
+
+* Tensorflow 2.3.0
+* Tensorflow probability 0.11.0
+
+References:
+[Density estimation using Real NVP](https://arxiv.org/pdf/1605.08803.pdf)
+"""
+
+"""
+## Setup
+
+"""
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import layers
+from tensorflow.keras import regularizers
+from sklearn.datasets import make_moons
+import numpy as np
+import matplotlib.pyplot as plt
+import tensorflow_probability as tfp
+
+"""
+## Load the data
+"""
+
+data = make_moons(3000, noise=0.05)[0].astype("float32")
+norm = layers.experimental.preprocessing.Normalization()
+norm.adapt(data)
+normalized_data = norm(data)
+
+"""
+## Affine coupling layer
+"""
+
+# Creating a custom layer with keras API.
+output_dim = 256
+reg = 0.01
+
+
+def Coupling(input_shape):
+    input = keras.layers.Input(shape=input_shape)
+
+    t_layer_1 = keras.layers.Dense(
+        output_dim, activation="relu", kernel_regularizer=regularizers.l2(reg)
+    )(input)
+    t_layer_2 = keras.layers.Dense(
+        output_dim, activation="relu", kernel_regularizer=regularizers.l2(reg)
+    )(t_layer_1)
+    t_layer_3 = keras.layers.Dense(
+        output_dim, activation="relu", kernel_regularizer=regularizers.l2(reg)
+    )(t_layer_2)
+    t_layer_4 = keras.layers.Dense(
+        output_dim, activation="relu", kernel_regularizer=regularizers.l2(reg)
+    )(t_layer_3)
+    t_layer_5 = keras.layers.Dense(
+        2, activation="linear", kernel_regularizer=regularizers.l2(reg)
+    )(t_layer_4)
+
+    s_layer_1 = keras.layers.Dense(
+        output_dim, activation="relu", kernel_regularizer=regularizers.l2(reg)
+    )(input)
+    s_layer_2 = keras.layers.Dense(
+        output_dim, activation="relu", kernel_regularizer=regularizers.l2(reg)
+    )(s_layer_1)
+    s_layer_3 = keras.layers.Dense(
+        output_dim, activation="relu", kernel_regularizer=regularizers.l2(reg)
+    )(s_layer_2)
+    s_layer_4 = keras.layers.Dense(
+        output_dim, activation="relu", kernel_regularizer=regularizers.l2(reg)
+    )(s_layer_3)
+    s_layer_5 = keras.layers.Dense(
+        2, activation="tanh", kernel_regularizer=regularizers.l2(reg)
+    )(s_layer_4)
+
+    return keras.Model(inputs=input, outputs=[s_layer_5, t_layer_5])
+
+
+Coupling(2).summary()
+"""
+## Real NVP
+"""
+
+
+class RealNVP(keras.Model):
+    def __init__(self, num_coupling_layers):
+        super(RealNVP, self).__init__()
+
+        self.num_coupling_layers = num_coupling_layers
+
+        # Distribution of the latent space.
+        self.distribution = tfp.distributions.MultivariateNormalDiag(
+            loc=[0.0, 0.0], scale_diag=[1.0, 1.0]
+        )
+
+        self.masks = np.array(
+            [[0, 1], [1, 0]] * (num_coupling_layers // 2), dtype="float32"
+        )
+
+        self.loss_tracker = keras.metrics.Mean(name="loss")
+
+        self.layers_list = [Coupling(2) for i in range(num_coupling_layers)]
+
+    @property
+    def metrics(self):
```

Comment:
Please include the docstring I provided, it's important for readers to understand why this method exists.